### PR TITLE
Add per-browser setting to ignore non-fatal dnsmasq warnings

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Add per-browser setting to ignore non-fatal `dnsmasq` warnings. This has the advantage that a user can decide on a per-device basis if they want to see these warnings (e.g. on a development machine) or not (e.g. on a tablet)

**How does this PR accomplish the above?:**

Add a per-browser setting in local storage and use its value to ignore `dnsmasq` warnings when querying the database.

PS: If you need to add warnings you can do the following:

1. Cause a regex warning by adding an invalid regex such as `[[][]][[][[[[`
2. Cause a non-fatal `dnsmasq` warning by running `pihole-FTL dhcp-discover` when Pi-hole is your DHCP server (will query `lo` which, typically, isn't configured for DHCP).
3. If your Pi-hole is not your DHCP server, set a ridiculously large cache size (like 20,000) to trigger a warning

@rdwebdesign I'm aware you're modifying the `checkMessages()` function in another PR too, one of us will have to merge the respective changes depending on which PR gets merged earlier. It should be simple to do.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: settings.php [hunk 1]
```diff
@@ -1217,6 +1217,14 @@ function convertseconds($argument)
                                             </div>
                                         </div>
                                     </div>
+                                      <div class="row">
+                                        <div class="col-md-12">
+                                            <div>
+                                                <input type="checkbox" name="hideNonfatalDnsmasqWarnings" id="hideNonfatalDnsmasqWarnings" value="no">
+                                                <label for="hideNonfatalDnsmasqWarnings"><strong>Hide non-fatal <code>dnsmasq</code> warnings</strong></label>
```
Comment (yubiuser) [inline settings.php]: Adding a link to https://docs.pi-hole.net/ftldns/dnsmasq_warn/ here?

**Review by yubiuser — COMMENTED** [review]
Checking/un-checking the checkbox does reduce the counter of the triangle as expected. However, if there are only non-fatal `dnsmasq` warnings, I would expect the triangle to be gone right after checking the box. But this does not happen - only a reload of the page removes the triangle completely.

**Comment by DL6ER** [comment]
Ah, yes. That's because there is only code to change the number, but not to remove the icon dynamically. I'll add this.

**Comment by DL6ER** [comment]
Waiting for https://github.com/pi-hole/AdminLTE/pull/2133 to get merged first, There will be merge conflicts afterwards.

**Comment by rdwebdesign** [comment]
Note:
#2133 was merged.

**Review by yubiuser — APPROVED** [review]
Works as intended.

**Pull request merged**